crates/common/tedge_config/src/tedge_config_cli/error.rs

```diff
@@ -29,6 +29,9 @@ pub enum TEdgeConfigError {
 
     #[error(transparent)]
     DirNotFound(#[from] tedge_utils::paths::PathsError),
+
+    #[error(transparent)]
+    FromReadError(#[from] crate::tedge_config_cli::new::ReadError),
```

Comment:
This error case might  be useless once all the uses of `ReadableKey` kept private to `tedge_config` thanks to helper methods as  `NewTEdgeConfig::c8y_host()` .

Reply:
The new version is better because only the very specific `C8yHttpConfigBuildError` error type transparently handles the `ReadError` case.